federation/pkg/kubefed/init/init.go

```diff
@@ -229,7 +230,25 @@ func initFederation(cmdOut io.Writer, config util.AdminConfig, cmd *cobra.Comman
 		return err
 	}
 
+	//to talk to federation ctrl plane
+	fedFactory := config.HostFactory(initFlags.Name, initFlags.Kubeconfig)
```

Comment:
> Also, I am somewhat worried about doing this. HostFactory() method was written keeping interactions with host cluster in mind. It wasn't for federation API server. Also, the discovery interface for the command factory builds the objects for Kubernetes APIs and not federation APIs. All of these might work today because of the particular way things are hooked up, but might break down when some of these assumptions are violated.
> 
> In order to avoid this problem in the future and also to do the right thing, directly build the federation Clientset in k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset. You can then call fedClientset.Discovery() to get the federation discovery client.

I have a slight confusion about this. Please help me understand better.
Actually when I implemented it first, I did something similar what are asking for. I did extend the AdminConfig interface to add another method getconfig, which built the config on similar lines as is currently done to get hostfactory, and then got a separate client for fed control plane from that config. I however removed that in favor of simplicity of the current one.
Also on futher checking, the HostFactory() impl gets the new factory from package kubectl/cmd which I guess "should" work with either federation API server or kube API server, for that particular version alike.

I would any ways keep the impl as suggested by you also ready, and update this commit or not, based on your suggestion.